hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/EntitiesAndCollectionsInSameRegionTest.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.cache.infinispan.functional;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.Hibernate;
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.cache.internal.DefaultCacheKeysFactory;
+import org.hibernate.cfg.Environment;
+import org.hibernate.stat.SecondLevelCacheStatistics;
+import org.hibernate.stat.Statistics;
+
+import org.hibernate.testing.TestForIssue;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * @author Gail Badner
+ */
+public class EntitiesAndCollectionsInSameRegionTest extends SingleNodeTest {
+	private static final String REGION_NAME = "ARegion";
+
+	private final AnEntity anEntity;
+	private final AnotherEntity anotherEntity;
+
+	public EntitiesAndCollectionsInSameRegionTest() {
+		anEntity = new AnEntity();
+		anEntity.id = 1;
+		anEntity.values.add( "abc" );
+
+		anotherEntity = new AnotherEntity();
+		anotherEntity.id = 1;
+		anotherEntity.values.add( 123 );
+	}
+
+	@Override
+	public List<Object[]> getParameters() {
+		return getParameters( true, false, false, false );
+	}
+
+	@Override
+	public Class[] getAnnotatedClasses() {
+		return new Class[] {
+				AnEntity.class,
+				AnotherEntity.class
+		};
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	protected void addSettings(Map settings) {
+		super.addSettings( settings );
+		settings.put( Environment.CACHE_KEYS_FACTORY, DefaultCacheKeysFactory.SHORT_NAME );
```

Comment:
This should be default in master, isn't it?

Reply:
The test uses a slightly different value:

    settings.put( Environment.CACHE_KEYS_FACTORY, DefaultCacheKeysFactory.SHORT_NAME )

I'm testing to see if the test demands this.

Reply:
If removed it fails with:

    org.hibernate.WrongClassException: Object [id=1] was not of the specified subclass...